src/header/index.ts

```diff
@@ -0,0 +1,198 @@
+    };
+    exp: number;
+    iat: number;
+    iss: string;
+    jti: string;
+    nonce: string;
+    sub: string;
+}
+export interface SolidSession {
+    authorization: SolidAuthorization;
+    credentialType: string;
+    idClaims: SolidClaim;
+    idp: string;
+    issuer: string;
+    sessionKey: string;
+    webId: string;
+}
+
+export async function initHeader (store: IndexedFormula) {
+  const header = document.getElementById('PageHeader')
+  if (!header) {
+    return
+  }
+
+  const pod = getPod()
+  solidAuthClient.trackSession(rebuildHeader(header, store, pod))
+}
+
+function rebuildHeader (header: HTMLElement, store: IndexedFormula, pod: NamedNode) {
+  return async (session: SolidSession | null) => {
+    const user = session ? sym(session.webId) : null
+    log(user)
+    header.innerHTML = ''
+    header.appendChild(await createBanner(store, pod, user))
+  }
+}
+
+async function createBanner (store: IndexedFormula, pod: NamedNode, user: NamedNode | null): Promise<HTMLElement> {
+  const podLink = document.createElement('a')
+  podLink.href = pod.uri
+  podLink.classList.add('header-banner__link')
+  podLink.innerHTML = icon
+
+  const menu = user
+    ? await createUserMenu(store, user)
+    : createLoginSignUpButtons()
+
+  const banner = document.createElement('div')
+  banner.classList.add('header-banner')
+  banner.appendChild(podLink)
+  banner.appendChild(menu)
+
+  return banner
+}
+
+function createLoginSignUpButtons () {
+  const profileLoginButtonPre = document.createElement('div')
+  profileLoginButtonPre.classList.add('header-banner__login')
+  profileLoginButtonPre.appendChild(loginStatusBox(document, null, {}))
+  return profileLoginButtonPre
+}
+
+async function openDashboardPane (outliner: any, pane: string): Promise<void> {
+  outliner.showDashboard({
+    pane
+  })
+}
+
+function createUserMenuButton (label: string, onClick: EventListenerOrEventListenerObject): HTMLElement {
+  const button = document.createElement('button')
+  button.classList.add('header-user-menu__button')
+  button.addEventListener('click', onClick)
+  button.innerText = label
+  return button
+}
+
+function createUserMenuLink (label: string, href: string): HTMLElement {
+  const link = document.createElement('a')
+  link.classList.add('header-user-menu__link')
+  link.href = href
+  link.innerText = label
+  return link
+}
+
+async function createUserMenu (store: IndexedFormula, user: NamedNode): Promise<HTMLElement> {
+  const fetcher = (<any>store).fetcher
+  if (fetcher) {
+    // Making sure that Profile is loaded before building menu
+    await fetcher.load(user)
+  }
+  // const outliner = getOutliner(document)
+
+  const loggedInMenuList = document.createElement('ul')
+  loggedInMenuList.classList.add('header-user-menu__list')
+  loggedInMenuList.appendChild(createUserMenuItem(createUserMenuLink('Show your profile', user.uri)))
+  /* const menuItems = await getMenuItems(outliner)
+  menuItems.forEach(item => {
```

Comment:
I was thinking we should extract this list into something that is provided from mashlib? Part of next step you're working on?

Reply:
yeah this is the next time with the options.  it's actually in the add-header branch now.